### DISCUSSION TITLE
Why doesn't this project utilize mimirtool?

### DISCUSSION THREAD

**Discussion by Itaykal** [grafana/grizzly#433] [Q&A]
I saw that cortextool was once used in this, but why was it removed and why isn't mimirtool used instead?
I believe it will reduce a lot of work on the mimir side of this tool...

**Comment by malcolmholmes** [comment]
Interacting with mimirtool as a CLI adds a dependency and is complicated in terms of testing/etc. Working directly with the Mimir API is cleaner, results in less code and is easier to test.

Does the new Mimir code cause you a problem?

**Reply by Itaykal** [reply]
It just looks like Grizzly tries to immitate some of mimirtool's features, this in my opinion adds a lot of code complexity as the project must manage it's own Mimir API client and/or develop features that are already pressent and "tried and tested" in mimirtool.

I think the best case scenario is to use the same client that mimirtool has as it's quite redundant to develop the same client for the same tasks. And thus enjoying the contributed features there as an optional enrichments to grizzly.

**Reply by malcolmholmes** [reply]
Apologies for lateness. We do use the Mimir golang client. And our aim is to be broader than just Mimir (mimirtool being focused on one product). Also, aiming for ease of use, requiring users to download a new tool would make life harder for them.
